sphinx/builders/_epub_base.py

```diff
@@ -167,6 +168,17 @@ def init(self) -> None:
         self.use_index = self.get_builder_config('use_index', 'epub')
         self.refnodes: list[dict[str, Any]] = []
 
+    def merge_builder_post_transform(self, new_attrs: dict[str, Any]) -> None:
```

Comment:
AFAICT, I think *new_attrs* are not meant to be writable so a mapping should be fine. 

By the way, "new_attrs" is a bit confusing for me. Maybe "context" or "extras" could be a better name.